### PR TITLE
tests: add Qwen3-TTS offline inference tests (happy + negative path)

### PR DESCRIPTION
**Pull request by Shaun-Walsh**
## Summary

- Add 11 new offline inference tests for both Qwen3-TTS models (Base 0.6B and CustomVoice 1.7B)
- Tests cover happy-path generation (special characters, acronyms, Chinese text, typos, long input) and negative-path validation (empty input, whitespace-only, missing ref_audio, missing ref_text)
- Negative-path testing exposed 3 offline pipeline validation gaps where the pipeline silently accepts invalid input that the HTTP serving layer correctly rejects

**CustomVoice** (`test_qwen3_tts_customvoice.py`) — 6 new tests:
| Test | Type | Status |
|------|------|--------|
| `test_customvoice_special_characters` | Happy path | PASSED |
| `test_customvoice_acronyms_and_abbreviations` | Happy path | PASSED |
| `test_customvoice_chinese_text` | Happy path | PASSED |
| `test_customvoice_rejects_empty_input` | Negative | PASSED |
| `test_customvoice_rejects_whitespace_only_input` | Negative | XFAIL (#3706) |
| `test_customvoice_handles_very_long_input` | Robustness | PASSED |

**Base** (`test_qwen3_tts_base.py`) — 5 new tests:
| Test | Type | Status |
|------|------|--------|
| `test_base_voice_clone_special_characters` | Happy path | PASSED |
| `test_base_voice_clone_with_typos` | Happy path | PASSED |
| `test_base_rejects_empty_input` | Negative | PASSED |
| `test_base_rejects_missing_ref_audio` | Negative | XFAIL (#3707) |
| `test_base_rejects_missing_ref_text` | Negative | XFAIL (#3708) |

**Upstream issues filed:**
- #3706 — Offline pipeline accepts whitespace-only input
- #3707 — Offline pipeline accepts missing ref_audio for Base task
- #3708 — Offline pipeline accepts missing ref_text for Base task

## Test plan

- [x] All 13 tests validated on H100 (`l42-h06-000-xe8640.rdu3.labs.perfscale.redhat.com`)
- [x] 10 passed, 3 xfailed (known upstream validation gaps)
- [x] Existing `test_text_to_audio_001` in both files unaffected
- [x] `ruff format` and `ruff check` pass
- [x] All commits GPG signed

```
CUDA_VISIBLE_DEVICES=2 pytest tests/e2e/offline_inference/test_qwen3_tts_customvoice.py tests/e2e/offline_inference/test_qwen3_tts_base.py -v --tb=short
```

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Review by hsliuustc0106 — COMMENTED** [review]
Good test coverage for both happy and negative paths. XFAIL markers properly document known validation gaps.

File: tests/e2e/offline_inference/test_qwen3_tts_base.py [hunk 1]
```diff
@@ -88,3 +88,111 @@ def test_text_to_audio_001(omni_runner, omni_runner_handler) -> None:
...
+
+@pytest.mark.advanced_model
+@pytest.mark.tts
+@hardware_test(res={"cuda": "L4"}, num_cards=1)
+@pytest.mark.parametrize("omni_runner", tts_server_params, indirect=True)
+def test_base_voice_clone_special_characters(omni_runner, omni_runner_handler) -> None:
+    """
+    Test voice cloning with special characters and punctuation in input text.
+    Deploy Setting: qwen3_tts_no_async_chunk.yaml + enforce_eager=true
+    Input Modal: text with special characters (!,$,—,...)
```
Comment (alex-jw-brooks) [inline tests/e2e/offline_inference/test_qwen3_tts_base.py]: nit: docstrings should be `modality` instead of `modal`

File: tests/e2e/offline_inference/test_qwen3_tts_customvoice.py [hunk 2]
```diff
@@ -77,3 +77,113 @@ def test_text_to_audio_001(omni_runner, omni_runner_handler) -> None:
...
+@pytest.mark.parametrize("omni_runner", tts_server_params, indirect=True)
+def test_customvoice_rejects_empty_input(omni_runner, omni_runner_handler) -> None:
+    """
+    Negative test: empty input text should be rejected.
+    Deploy Setting: qwen3_tts_no_async_chunk.yaml + enforce_eager=true
+    Input Modal: empty string
+    Expected: ValueError or RuntimeError
+    """
+    request_config = {"input": "", "voice": "vivian"}
+    with pytest.raises((ValueError, RuntimeError)):
```
Comment (alex-jw-brooks) [inline tests/e2e/offline_inference/test_qwen3_tts_customvoice.py]: Since the error type should be deterministic, I think the `pytest.raises` calls should just have the error type that is actually raised per test

File: tests/e2e/offline_inference/test_qwen3_tts_customvoice.py [hunk 3]
```diff
@@ -77,3 +77,113 @@ def test_text_to_audio_001(omni_runner, omni_runner_handler) -> None:
...
+
+
+@pytest.mark.advanced_model
+@pytest.mark.tts
+@hardware_test(res={"cuda": "L4"}, num_cards=1)
+@pytest.mark.parametrize("omni_runner", tts_server_params, indirect=True)
+def test_customvoice_chinese_text(omni_runner, omni_runner_handler) -> None:
+    """
+    Test text-to-audio with Chinese language text.
+    Deploy Setting: qwen3_tts_no_async_chunk.yaml + enforce_eager=true
```
Comment (alex-jw-brooks) [inline tests/e2e/offline_inference/test_qwen3_tts_customvoice.py]: I think `Deploy Setting:` can be removed from the docstrings in these tests, since they are all using the same params

**Review by alex-jw-brooks — CHANGES_REQUESTED** [review]

File: tests/e2e/offline_inference/test_qwen3_tts_base.py (hunk 1 above)
Comment (Shaun-Walsh) [reply tests/e2e/offline_inference/test_qwen3_tts_base.py]: Thanks Alex, changed.

File: tests/e2e/offline_inference/test_qwen3_tts_customvoice.py (hunk 2 above)
Comment (Shaun-Walsh) [reply tests/e2e/offline_inference/test_qwen3_tts_customvoice.py]: Changed to reflect the error type

File: tests/e2e/offline_inference/test_qwen3_tts_customvoice.py (hunk 3 above)
Comment (Shaun-Walsh) [reply tests/e2e/offline_inference/test_qwen3_tts_customvoice.py]: Fixed

**Comment by oglok** [comment]
@alex-jw-brooks @hsliuustc0106 is it good to be merged?

File: tests/e2e/offline_inference/test_qwen3_tts_customvoice.py [hunk 4]
```diff
@@ -69,11 +69,114 @@ def get_prompt():
...
+    """
+    Test text-to-audio with acronyms and abbreviations.
+    Input Modality: text with e.g., i.e., Dr., p.m., acronyms
+    Output Modality: audio
+    """
+    request_config = {
+        "input": "The CEO of NASA, e.g. Dr. Smith, will arrive at 3 p.m. i.e. before the Q&A.",
+        "voice": "vivian",
+    }
+    omni_runner_handler.send_audio_speech_request(request_config)
```
Comment (alex-jw-brooks) [inline tests/e2e/offline_inference/test_qwen3_tts_customvoice.py]: I think this implies that it's more of a functional test than it is, since we don't actually inspect the output. If these are meant to mostly be smoke tests to make sure we get a valid response out, maybe it would be better to just collapse the overlapping ones (e.g., this and the one above it since they are similar with different input strings) into one test and `mark.parametrize` over the test strings to make sure they all give some response back.

If the intention is to validate that this is actually handling things like acronyms correctly, it should probably be run with deterministic sampling params and the output should be inspected for correctness I think

**Review by alex-jw-brooks — CHANGES_REQUESTED** [review]
Thanks @oglok @Shaun-Walsh! One comment about the intention of these tests that would be nice to clarify first, but otherwise looks reasonable to me.

Also, since some of these tests are things like long sequences, can you please post a comment with the runtime of the tests on `main` vs this branch so that we know how much impact it may have on the CI?

File: tests/e2e/offline_inference/test_qwen3_tts_customvoice.py (hunk 4 above)
Comment (Shaun-Walsh) [reply tests/e2e/offline_inference/test_qwen3_tts_customvoice.py]: Hey hey. Yep, you're righ they're smoke tests. Collapsed them into parametrized tests:
- test_customvoice_smoke — [special-characters, acronyms, chinese-text, long-input]
- test_base_voice_clone_smoke — [special-characters, disfluent-text]
 
Negative tests left as-is.
Ran the timing 3 times on an L4 to be sure. Main sits at ~322s for the 2 existing tests, this branch comes in at ~852s for 13. So about +530s extra, but no additional model load cost since the parametrized tests share the fixture. Most of the time is the CustomVoice generation tests which are ~127s a pop.

**Comment by Shaun-Walsh** [comment]
Hey @alex-jw-brooks — just circling back on this one. I believe all your feedback has been addressed:

- Docstrings fixed (`modal` → `modality`)
- `pytest.raises` now uses specific error types per test
- `Deploy Setting:` removed from docstrings
- Collapsed overlapping tests into parametrized smoke tests
- Posted L4 timing data (main ~322s → branch ~852s, +530s with no additional model load cost)

This PR is a dependency for our Red Hat Midstream build of vllm-omni. Could you take another look when you get a chance? Happy to make any further adjustments quickly. Thanks!

File: tests/e2e/offline_inference/test_qwen3_tts_customvoice.py (hunk 4 above)
Comment (Shaun-Walsh) [reply tests/e2e/offline_inference/test_qwen3_tts_customvoice.py]: Hey @alex-jw-brooks just following up on getting this on reviewed and merged
